[CS2103T-T11-1] MooLah

briyanii opened the pull request:
@anqichen9856
@czkay
@qweiping31415
@ryoarmanda

DivineDX commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +311,99 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 |=======================================================================
 |Priority |As a ... |I want to ... |So that I can...
-|`* * *` |new user |see usage instructions |refer to instructions when I forget how to use the App
+|`* * *` |full-time university student without any stable income |record my expenses |track my spending and practice financial management skills
```
User doesn't seem specific: The need to record expenses is not exclusive to a "full-time university student without any stable income"

DivineDX commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +311,99 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
-|`* * *` |new user |see usage instructions |refer to instructions when I forget how to use the App
+|`* * *` |full-time university student without any stable income |record my expenses |track my spending and practice financial management skills
+
+|`* * *` |non-money minded person |view a list of all my previous expenses |be better at tracking my expenses
```
User seems wrong. Should it be "a money minded person" instead?

DivineDX commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +311,99 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* * *` |non-money minded person |view a list of all my previous expenses |be better at tracking my expenses
+
+|`* * *` |user |categorize my expenses by types (e.g. Transport, Food, etc.) |differentiate expenses easily
```
'user' could be more specific

eshwarkp commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,25 +311,90 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 _{More to be added}_
 
 [appendix]
 == Use Cases
```
*Use Cases Review*

1) Step numbering is accurate. Well Done!
2) Some possible extensions mentioned. Well Done!
3) No unnecessary UI details mentioned. Well Done!
4) Formatting and notations are accurate. Well Done!

DivineDX commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +311,99 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* * *` |YEP (Youth Expedition Project) organizer who is in charge of logistics for my overseas project |delete a previous expense |I do not keep track of expenses that have been reimbursed by Youth Corps Singapore
+
+|`* * *` |non-money minded person |save the expenses data locally |still keep track of my expenses even after closing the App
```
User seems wrong. Should it be "a money minded person" instead?

calvincxz commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,25 +311,90 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+(For all use cases below, the *System* is `NUSpend` and the *Actor* is the `user`, unless specified otherwise)
 
 [discrete]
 === Use case: Delete person
```
is Delete person still a relevant command?

DivineDX commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +311,99 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* *` |money-minded user |extrapolate from daily/weekly expenses to monthly expenses |have a rough gauge on future spending and plan budgets accordingly
+
+|`* *` |user |manually input my expenses in an external file and import these expenses into the App |track previous expenses that occur before I start using the App
```
'user' could be more specific

DivineDX commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +311,99 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* *` |user |manually input my expenses in an external file and import these expenses into the App |track previous expenses that occur before I start using the App
+
+|`* *` |user |request for periodic statements |revise and reflect on my past expenses
```
'user' could be more specific

DivineDX commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +311,99 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* *` |visual person who easily gets confused by a large amount of text and prefers graphic illustrations |view my monthly expenses in pie charts |better understand the weightage of spending in different categories in a more direct and efficient manner
+
+|`* *` |user |update details of a past expense |correct erroneous past expenses without having to delete it and add again
```
'user' could be more specific

DivineDX commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +311,99 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* *` |careless person who is prone to typos |redo my previous command |I do not delete expenses by accident
+
+|`* *` |non-money minded person |set periodic budgets |discipline my spending
```
User seems wrong. Should it be "a money minded person" instead?

DivineDX commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +311,99 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* *` |non-money minded person |set periodic budgets |discipline my spending
+
+|`* *` |non-money minded person |set categorical budgets |exceptional expenses (e.g. healthcare or sponsored expenses) are not tracked under default budget to prevent confusion
```
User seems wrong. Should it be "a money minded person" instead?

DivineDX commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +311,99 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* *` |non-money minded person |set categorical budgets |exceptional expenses (e.g. healthcare or sponsored expenses) are not tracked under default budget to prevent confusion
+
+|`* *` |non-money minded person |let the App suggest some budgeting tips for me |be better at tracking my expenses
```
User seems wrong. Should it be "a money minded person" instead?

calvincxz reviewed (commented):
Good job!

DivineDX commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +311,99 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+|`*` |forgetful person |receive email reminders if I haven’t input my expenses for a few days |do not forget to track my expense
 
-|`* * *` |user |delete a person |remove entries that I no longer need
+|`*` |someone who spends a lot of time in public spaces in school |easily hide my budgeting information on the application quickly |others could not see the details on my screen
```
User could perhaps be a better phrased as "Someone who values privacy"

DivineDX commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +311,99 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`*` |non-financially independent student |hide some details when exporting expenses |do not share exact details with my parents but instead include them as an anonymous expense
+
+|`*` |non-money minded person |let the App help me make saving plans for future expenses |start saving early
```
User seems wrong. Should it be "a money minded person" instead?

DivineDX commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +311,99 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`*` |non-money minded person |let the App help me make saving plans for future expenses |start saving early
+
+|`*` |user |group certain expenses, e.g. group chicken rice and milo as lunch, but both remain separate entries as well |manage related expenses together
```
'user' could be more specific

DivineDX commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,35 +311,99 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`*` |advanced CLI user |add aliases to my common inputs |avoid typing the similar things every time
+
+|`*` |user |let the App predict frequent similar expenses and prompt me to enter them |do not miss expenses while I'm not using the App
```
'user' could be more specific

DivineDX reviewed (commented):
Tutorial: PR Review

DivineDX commented on docs/DeveloperGuide.adoc:
```diff
@@ -153,89 +164,250 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+
+The following sequence diagram shows how the add budget operation works:
+
+image::AddBudgetSequenceDiagram.png[]
```
Is there still a class called "AddressBookParser" and "AddressBook"? I believe with the refactor they should not be present anymore

the last method call in the diagram:
> setPrimary(toAdd)

is an object calling a method of itself. It is of the wrong notation.

DivineDX commented on docs/DeveloperGuide.adoc:
```diff
@@ -153,89 +164,250 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+
+The following activity diagram summarizes what happens when the app launches and when a new expense is added:
+
+image::BudgetActivityDiagram.png[]
```
Diagram too big. Considering editing the code to resize.

> image::BudgetActivityDiagram.png[BudgetActivityDiagram,295,607]

Where the first parameter is a "alt-text" to be displayed if the image cannot be found, second parameter is the width (pixels), third parameter is the height (pixels)

Otherwise, the diagram looks fairly well done.

DivineDX commented on docs/DeveloperGuide.adoc:
```diff
@@ -153,89 +164,250 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+** Cons: User would not be able to track expenses that fall under multiple Budgets.
+* **Alternative 2:** Allow expenses without a Budget.
+** Pros: More flexible budgeting experience.
+** Cons: Internal logic would be very confusing. GUI and statistics would be harder to implement.
```
Perhaps can elaborate more on why it is very confusing

shihaoyap reviewed (commented):
Budget Feature Feedback
1) Budget Object Diagram -> Pretty details, however there is an unknown symbol C, A. will be good to explain the notations.
2) Alot of object diagrams included -> which is good, but i think it can be improved by separating the diagrams to different sub sections. 
3) AddBudgetCommand and SwitchBudgetCommand -> are kind of similar, might want to consider just using one of that diagram. Since part of the marking scheme says that you might be marked down for too many unnecessary diagrams.
4) Descriptions are in chunks, can consider breaking it down to sub sections for better readability.

SoilingRogue commented on docs/DeveloperGuide.adoc:
```diff
@@ -153,89 +164,263 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
-Step 1. The user launches the application for the first time. The `VersionedAddressBook` will be initialized with the initial address book state, and the `currentStatePointer` pointing to that single address book state.
+Step 1. The user launches the application for the first time. The current `ModelHistory` is now empty.
 
 image::UndoRedoState0.png[]
```
Just from this image alone, the relationship between current state and model history is not clearly illustrated, maybe you can have another bigger State box covering both m0:Model and the ModelHistory box to show that model history contains states which are not current state, and not just simply floating around.

SoilingRogue commented on docs/DeveloperGuide.adoc:
```diff
@@ -153,89 +164,263 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+* isPrimary: `boolean` - a flag indicating the primary status of this budget
+* proportionUsed: `Percentage` - the proportion of the total amount that has been used up
+
+image::BudgetClassDiagram.png[]
```
Unnecessarily in depth diagram of the budget class

charliechoong commented on docs/DeveloperGuide.adoc:
```diff
@@ -153,89 +164,263 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+will be tracked under this budget. If the user wants to track an expense under a different budget, he
+switches to the target budget first, before adding the expense.
+
+Each `Budget` has 8 fields:
```
We see that your bullet points are not working

charliechoong reviewed (commented):
Well done with the diagrams.

SoilingRogue commented on docs/DeveloperGuide.adoc:
```diff
@@ -153,89 +164,263 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+
+The following sequence diagram shows how the `stats` operation works:
+
+image::BasicStatisticsCommand.png[]
```
The sequence diagram is too tiny, difficult to view. Maybe you can start from the portion where StatsCommand is called, as the AddressBookParser, StatsParser, tokenizer etc are applicable for all commands.

SoilingRogue reviewed (commented):
Great job overall

briyanii committed:
Test for GenericCommandWord

briyanii committed:
Test for DeleteAliasCommandParser

briyanii committed:
Tests for deleteAliasCommandParser

briyanii committed:
Add test for MooLah parser

briyanii committed:
Riase MooLahParser coverage to 100%

briyanii committed:
Fix checkstyle violations

briyanii committed:
Changed test

briyanii committed:
Merge pull request #247 from briyanii/tests

More Tests for Alias related packages, GenericCommands, and MooLahParser

briyanii committed:
Fix new CommandGroup causing generic command add to not work properly

briyanii committed:
Merge pull request #252 from briyanii/fix_5

Fix new CommandGroup causing generic command add to not work properly

briyanii committed:
Remove unused constructors

anqichen9856 committed:
Rename seedu.address to seedu.moolah and re-arrange the packages in /main and /test

Fix bug: edit & delete expense from budget not undoable

anqichen9856 committed:
Merge pull request #253 from anqichen9856/master

Rename and rearrange packages

briyanii committed:
Ui Test for CommandTextField

briyanii committed:
Remove unneeded files

qweiping31415 committed:
Populate SampleUtil for Product Demo (#254)

* Remove some javadoc comments

* Test Statistics class in model

* Remove expense parameter for calculateStats

* Add tests for PieChartStatistics

* Add tests to TabularStatistics

* Rename variables and improve conditional statements

* Populate TypicalMooLah with sample data

anqichen9856 committed:
Add unit tests for budget-related help classes, commands and command parsers

anqichen9856 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main

* 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main:
  Populate SampleUtil for Product Demo (#254)

anqichen9856 committed:
Rename getter methods in Budget and BudgetWindow for better readability

- Budget#getStartDate() -> Budget#getWindowStartDate()
- Budget#getEndDate() -> Budget#getWindowEndDate()
- Budget#getPeriod() -> Budget#getBudgetPeriod()
- BudgetWindow#getPeriod() -> BudgetWindow#getBudgetPeriod()

anqichen9856 committed:
Add assertion in unit test to check precondition

anqichen9856 committed:
Fix checkstyle violation in tests

anqichen9856 committed:
Comment out run_allFieldsSpecified_success() in EditExpenseFromBudgetCommandTest since it passed locally but failed Travis

anqichen9856 committed:
Make toString() test in BudgetWindow accept US time format for Travis compatibility

anqichen9856 committed:
Fix description violations in SampleDataUtil

anqichen9856 committed:
Merge pull request #256 from anqichen9856/master

Add unit tests for budget-related classes

anqichen9856 committed:
Fix bug: undoing "add expense" not reflected in primary budget panel

anqichen9856 committed:
Change dates of some expenses in SampleDataUtil so that the current period budget is more populated

czkay committed:
Testing: More Event tests (#255)

* Add more Event tests

* Fix Expense not showing after accepting Events prompt popup while in Primary Budget panel

* Change datetime of all test expenses to past timestamps

* Merge branch 'master' into branch-Events

# Conflicts:
#	src/test/java/seedu/moolah/testutil/TypicalMooLah.java

* Fix unit tests

* Fix checkstyle errors

* Fix checkstyle error

ryoarmanda committed:
Add Coveralls Badge in readme (#258)

ryoarmanda committed:
Add menu prefix and implement AddMenuExpenseCommand (#257)

* Add canteen menu support on add expenses

* Implement AddMenuExpense command

* Fix residual error

* Add sample menu item list and update UG

* Fix EOF error

anqichen9856 committed:
Update Ui.png with annotations

anqichen9856 committed:
Update UG: Introduction, Budgeting, and Command Summary

anqichen9856 committed:
Update PPP: team tasks

anqichen9856 committed:
Move code related to tracking initial and final budget status from MainWindow to Logic

anqichen9856 committed:
Add validation for max budget percentage value in AddExpenseCommand

anqichen9856 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main

* 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main:
  Add menu prefix and implement AddMenuExpenseCommand (#257)
  Add Coveralls Badge in readme (#258)
  Testing: More Event tests (#255)

anqichen9856 committed:
Merge pull request #259 from anqichen9856/master

Budgeting: bug fixes, UG & PPP updates, code quality improvement

anqichen9856 committed:
Update version in MainApp

anqichen9856 committed:
Merge pull request #260 from anqichen9856/master

Update version in MainApp

anqichen9856 committed:
Fix bug: switch period not reflected in json sometimes

czkay committed:
Update Events popup window (#262)

anqichen9856 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main

* 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main:
  Update Events popup window (#262)

anqichen9856 committed:
Merge pull request #263 from anqichen9856/master

Fix bug: switch period not reflected in json sometimes

briyanii committed:
GUI Tests for text field

ryoarmanda committed:
Add AddMenuExpenseCommand to reserved command words (#264)

anqichen9856 committed:
Fix bug: budgets refreshing every 10 seconds, making "switchperiod" unable to maintain the current status

- Change budget refreshing implementation in Timekeeper such that budgets only refresh at day breaks (00:00)

briyanii committed:
Add more GUI test

qweiping31415 committed:
Merge pull request #265 from anqichen9856/master

Fix bug: "switchperiod" unable to maintain the status

briyanii committed:
Update AppParametersTest to test all lines in equal and hashcode.

briyanii committed:
Add Test for MainWindow.java

This test is work in progress and coverage may not be accurate

briyanii committed:
Changes to suggestion feature behaviour to reduce lag

briyanii committed:
Tests for CommandBox and related classes

briyanii committed:
Modify fxml to not show version warning

briyanii committed:
Fix checkstyle violations

briyanii committed:
Update tests

qweiping31415 committed:
Improve Code Quality for StatsCommand

qweiping31415 committed:
Improve Code Quality for StatsTrendCommand

qweiping31415 committed:
Made Statistics an interface

qweiping31415 committed:
Add tests for descriptor classes

qweiping31415 committed:
Update tests for refactored classes

qweiping31415 committed:
Merge pull request #267 from qweiping31415/codeQualityStats

Code quality stats

anqichen9856 committed:
Fix failing unit tests

anqichen9856 committed:
Fix bug: editing/deleting expense from expenseList not reflected in corresponding budgets

anqichen9856 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main

* 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main:
  Update tests for refactored classes
  Add tests for descriptor classes
  Made Statistics an interface
  Improve Code Quality for StatsTrendCommand
  Improve Code Quality for StatsCommand

anqichen9856 committed:
Fix previously commented out tests

anqichen9856 committed:
Increase test coverage for Budget part to 100%

qweiping31415 committed:
Merge pull request #268 from anqichen9856/master

Unit tests & bug fixes

qweiping31415 committed:
Update DG diagrams

briyanii committed:
Add GUI tests for MooLah

Copied gui tests files from AB4 and adapted where neccessary to suit MooLah

Refactor existing GUI tests where neccessary

qweiping31415 committed:
Add newline to UML files

briyanii committed:
Fix eof checkstyle error

qweiping31415 committed:
Merge pull request #269 from qweiping31415/updateDocs

Update DG diagrams

briyanii committed:
Update build.gradle to match AB4 for gui tests

briyanii committed:
Add some tests for event panel

briyanii committed:
Fix checkstyle violations

briyanii committed:
Change travis.yml

briyanii committed:
Remove tests which passes locally but fails over travis

briyanii committed:
Remove unused import

briyanii committed:
Attempt to fix failing travis build with update to travis.yml

briyanii committed:
Re-add removed tests and comment it out

briyanii committed:
Undo removed lines from travis

briyanii committed:
Merge pull request #271 from briyanii/gui-test

Added some GUI tests for MooLah and some modifications to suggestion behaviour

briyanii committed:
Fix alias message usage

briyanii committed:
Fix textfield being too short and some letters not displaying properly (e.g. g)

qweiping31415 committed:
Refactor StatsPanel to use factory classes

briyanii committed:
Replace 'person' with appropriate class

anqichen9856 committed:
Update DG: Budgeting

anqichen9856 committed:
Cosmetic changes (remove commented out code, change "Moolah" to "MooLah" in javadoc comments)

anqichen9856 committed:
Fix bug in BudgetWindow#normalize() for weekly budgets

anqichen9856 committed:
Change max price to 10^21

anqichen9856 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main

* 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main: (25 commits)
  Undo removed lines from travis
  Re-add removed tests and comment it out
  Attempt to fix failing travis build with update to travis.yml
  Remove unused import
  Remove tests which passes locally but fails over travis
  Change travis.yml
  Fix checkstyle violations
  Add some tests for event panel
  Update build.gradle to match AB4 for gui tests
  Fix eof checkstyle error
  Add GUI tests for MooLah
  Update tests
  Fix checkstyle violations
  Modify fxml to not show version warning
  Tests for CommandBox and related classes
  Changes to suggestion feature behaviour to reduce lag
  Add Test for MainWindow.java
  Update AppParametersTest to test all lines in equal and hashcode.
  Add more GUI test
  GUI Tests for text field
  ...

qweiping31415 committed:
Repackage StatsPanel

qweiping31415 committed:
Merge pull request #278 from qweiping31415/featureBranch

Refactoring of StatsPanel

anqichen9856 committed:
Fix failing tests by changing default budget limit in test json files

anqichen9856 committed:
Update PPP

anqichen9856 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main

* 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main:
  Repackage StatsPanel
  Refactor StatsPanel to use factory classes

anqichen9856 committed:
Fix checkstyle violations

czkay committed:
Testing: More Event tests (#274)

* Add more Event tests

* Add more Event tests

* Improve processing of transpired events

* Fix bug with undoing already transpired Events

anqichen9856 committed:
Update PPP: team tasks, and add skeleton for introduction

anqichen9856 committed:
Add new line at the end of puml files

anqichen9856 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main

* 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main:
  Testing: More Event tests (#274)

anqichen9856 committed:
Update Javadoc comment

anqichen9856 committed:
Comment out tests (in JsonAdaptedBudgetTest and JsonSerializableMooLahTest) that pass locally but fail Travis

anqichen9856 committed:
Remove unused imports in JsonAdaptedBudgetTest

anqichen9856 committed:
Fix checkstyle violations

anqichen9856 committed:
Update link to RepoSense in PPP

anqichen9856 committed:
Merge pull request #279 from anqichen9856/master

DG & PPP updates & bug fixes

briyanii committed:
Changes to UG

ryoarmanda committed:
Improve model history to store parts of model instead of whole

ryoarmanda committed:
Update Javadocs

qweiping31415 committed:
Update UML Diagrams for DG

anqichen9856 committed:
Update UG, DG, PPP

anqichen9856 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main

* 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main:

briyanii committed:
Merge pull request #276 from briyanii/alias-fix

Fix alias message usage

briyanii committed:
Merge pull request #277 from briyanii/ui-test-comments

Replace 'person' with appropriate class

briyanii committed:
Changes to Panel Manager interface and Updates to UG DG

anqichen9856 committed:
Merge pull request #282 from anqichen9856/master

Update UG, DG, PPP

briyanii committed:
Undo commenting out of code which causes panels to not refresh properly

briyanii committed:
Merge pull request #283 from briyanii/uncomment

Undo commenting out of code which causes panels to not refresh properly

qweiping31415 committed:
Add newline

briyanii committed:
Fix mainwindow such that undo is properly reflected

qweiping31415 committed:
Merge pull request #284 from qweiping31415/updatePPP

Update UML Diagrams for DG

briyanii committed:
Add EOF to files

qweiping31415 committed:
Insert brackets

briyanii committed:
Remove unused import

qweiping31415 committed:
Merge pull request #286 from qweiping31415/insertBracket

Insert brackets to DG

briyanii committed:
Update UG to list all generic command configurations

czkay committed:
Update UG, DG and PPP (#287)

* Update documentation

* Update PPP

* Update UG

* Update documentation

ryoarmanda committed:
Merge branch 'master' into undo-redo-changes-only

czkay committed:
Improve grammar of some error messages

briyanii committed:
Merge pull request #285 from briyanii/ugdg

Update to UGDG and fix for undo not refreshing

ryoarmanda committed:
Merge branch 'master' into undo-redo-changes-only

anqichen9856 committed:
Tidy up code, update Javadoc comments, remove dead code

ryoarmanda committed:
Fix minor error

czkay committed:
Move classes into more appropriate packages

* Also refactor duplicate lines in Timestamp class

czkay committed:
Fix checkstyle errors

czkay committed:
Fix checkstyle errors

anqichen9856 committed:
Update DG: Instructions for Manual Testing

anqichen9856 committed:
Add legends section in UG & DG

anqichen9856 committed:
Merge pull request #288 from czkay/branch-CodeQuality

Improve code quality

ryoarmanda committed:
Merge branch 'master' into undo-redo-changes-only

anqichen9856 committed:
Remove dead code

anqichen9856 committed:
Add preface to Quick Start sections in UG & DG

ryoarmanda committed:
Reference correct import in ModelChanges

anqichen9856 committed:
Fix bug in BudgetWindow#normalize

anqichen9856 committed:
Update UG & DG

anqichen9856 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main

* 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main:
  Fix checkstyle errors
  Fix checkstyle errors
  Move classes into more appropriate packages
  Update UG to list all generic command configurations
  Remove unused import
  Add EOF to files
  Fix mainwindow such that undo is properly reflected
  Changes to Panel Manager interface and Updates to UG DG
  Changes to UG
  Improve grammar of some error messages
  Update UG, DG and PPP (#287)
  Insert brackets
  Add newline
  Update UML Diagrams for DG
  Undo commenting out of code which causes panels to not refresh properly

# Conflicts:
#	docs/DeveloperGuide.adoc
#	docs/UserGuide.adoc
#	src/main/java/seedu/moolah/logic/parser/ParserUtil.java

qweiping31415 committed:
Merge pull request #281 from ryoarmanda/undo-redo-changes-only

Improve undo/redo to store parts of models instead of a whole copy

anqichen9856 committed:
Remove tests that pass locally but fail Travis

anqichen9856 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main

* 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main:
  Reference correct import in ModelChanges
  Fix minor error
  Update Javadocs
  Improve model history to store parts of model instead of whole

# Conflicts:
#	src/test/java/seedu/moolah/logic/commands/budget/DeleteBudgetByIndexCommandTest.java
#	src/test/java/seedu/moolah/logic/commands/budget/DeleteBudgetByNameCommandTest.java

anqichen9856 committed:
Merge pull request #289 from anqichen9856/master

Tidy up code, update Javadoc comments, remove dead code, update documentation

ryoarmanda committed:
Fix alias undo bug (#291)

* Fix alias undo bug

* Remove unused import

qweiping31415 committed:
Update PPP

czkay committed:
Delete dead code and improve JavaDocs

czkay committed:
Merge branch 'master' into branch-Documentation

qweiping31415 committed:
Merge pull request #292 from qweiping31415/updateUG

Update PPP

czkay committed:
Update documentation

ryoarmanda committed:
Update UG

ryoarmanda committed:
Fix horizontal line in UG

ryoarmanda committed:
Update AboutUs

anqichen9856 committed:
Add Javadoc comments to Budget-related methods

ryoarmanda committed:
Retitle menu items table

ryoarmanda committed:
[WIP] Update Documentation (#294)

* Update UG

* Fix horizontal line in UG

* Update AboutUs

* Retitle menu items table

anqichen9856 committed:
Update documentation

anqichen9856 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main

* 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main:
  [WIP] Update Documentation (#294)
  Update PPP
  Fix alias undo bug (#291)

# Conflicts:
#	docs/AboutUs.adoc

anqichen9856 committed:
Update diagrams to suit MooLah instead of AddressBook

qweiping31415 committed:
Remove comments

anqichen9856 committed:
Merge pull request #295 from anqichen9856/master

Update documentation: AboutUs, README, PPP, DG, UG

ryoarmanda committed:
Update portfolio

czkay committed:
Update documentation

czkay committed:
Update PPP

qweiping31415 committed:
class diagram

czkay committed:
Merge branch 'master' into branch-Documentation

czkay committed:
Fix checkstyle errors

qweiping31415 committed:
class diagram

ryoarmanda committed:
Update DG

briyanii committed:
Merge pull request #297 from ryoarmanda/update-documentation

[WIP] Update documentation: portfolio

qweiping31415 committed:
Update PPP

ryoarmanda committed:
Improve DG

qweiping31415 committed:
Merge pull request #296 from qweiping31415/updateUG

Remove comments

czkay committed:
Update JavaDoc and remove dead code (#293)

* Delete dead code and improve JavaDocs

* Update documentation

* Update documentation

* Update PPP

* Fix checkstyle errors

briyanii committed:
Pdf

briyanii committed:
Ugdg (#298)

* Update DG for changes to suggestion feature

* Update PPP and DG

* Update to dg ppp

ryoarmanda committed:
Update documentation (#299)

* Update DG

* Improve DG

ryoarmanda committed:
Merge branch 'master' into update-documentation

briyanii committed:
UGDG update

briyanii committed:
Fix EOF violation

briyanii committed:
Fix EOF error

briyanii committed:
Fix wrong plural commands

briyanii committed:
Merge pull request #300 from briyanii/late

Late

czkay committed:
Merge branch 'master' into branch-Documentation

briyanii committed:
Fix wrong example

czkay committed:
Merge branch 'master' into branch-Documentation

czkay committed:
Improve UML diagram formatting

briyanii committed:
Merge pull request #301 from briyanii/late2

Fix wrong example

anqichen9856 committed:
Update UG

anqichen9856 committed:
Merge pull request #302 from czkay/branch-Documentation

Update documentation

anqichen9856 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main

* 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main:
  Improve UML diagram formatting
  Fix wrong example
  Fix wrong plural commands
  Fix EOF error
  Fix EOF violation
  UGDG update
  Pdf
  Fix checkstyle errors
  Update PPP
  Update documentation
  Update documentation
  Delete dead code and improve JavaDocs

qweiping31415 committed:
Update UG

ryoarmanda committed:
Update DG further

anqichen9856 committed:
Merge pull request #303 from anqichen9856/master

Update UG

briyanii committed:
Update wrong javadocs

ryoarmanda committed:
Merge branch 'master' into update-documentation

anqichen9856 committed:
Merge pull request #304 from ryoarmanda/update-documentation

Update DG

ryoarmanda committed:
Fix minor error

anqichen9856 committed:
Merge pull request #306 from briyanii/late3

Update wrong javadocs

qweiping31415 committed:
UG

anqichen9856 committed:
Merge pull request #307 from ryoarmanda/update-documentation

Fix minor error in DG

anqichen9856 committed:
Merge pull request #305 from qweiping31415/latePR

Update UG

briyanii committed:
remove merge conflict line

briyanii committed:
Update model and logic diagram

briyanii committed:
Fix wrong example

briyanii committed:
Fix linebreak

briyanii committed:
Merge pull request #309 from briyanii/final

Updates to DG UG

ryoarmanda committed:
Update DG Latest (#310)

* Sync changes

* Update UG latest

* Clean up docs

briyanii committed:
Update future features

qweiping31415 committed:
Merge pull request #311 from briyanii/final2

Update future features

qweiping31415 committed:
Fixed bug on start date

qweiping31415 committed:
Merge pull request #312 from qweiping31415/newPR

Fixed bug on start date

czkay committed:
Fix bug with imprecise parsing of expenses and events (#313)

qweiping31415 committed:
More tests and update DG

qweiping31415 committed:
Merge pull request #315 from qweiping31415/moreTests

More tests and update DG

anqichen9856 committed:
Update documentation

anqichen9856 committed:
Add Javadoc comments

anqichen9856 committed:
Populate SampleDataUtil further

anqichen9856 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main

* 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main:
  More tests and update DG
  Fix bug with imprecise parsing of expenses and events (#313)
  Fixed bug on start date
  Update future features
  Update DG Latest (#310)
  Fix linebreak
  Fix wrong example
  Update model and logic diagram

anqichen9856 committed:
Merge pull request #316 from anqichen9856/master

Update documentation, write Javadoc & populate SampleDataUtil

ryoarmanda committed:
Update documentations (#317)

* Sync changes

* Update UG latest

* Clean up docs

* Update DG

* Fix links

* Remove pdf

* Fix error and remove html

* Fix grammar error

anqichen9856 committed:
Update UG, DG & PPP

czkay committed:
Update documentation (#314)

* Update documentation

* Update documentation

* Also fix bug related to editing an expense's timestamp to the future

anqichen9856 committed:
Update DG: test cases

anqichen9856 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main

* 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main:
  Update documentation (#314)
  Update documentations (#317)

# Conflicts:
#	docs/DeveloperGuide.adoc

briyanii committed:
Update DG

briyanii committed:
Update DG

anqichen9856 committed:
Update DG

anqichen9856 committed:
Update DG

qweiping31415 committed:
Update Documentation

anqichen9856 committed:
Update DG

briyanii committed:
Merge pull request #319 from briyanii/final3

Update DG

anqichen9856 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main

* 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main:
  Update DG
  Update DG

qweiping31415 committed:
Updated Sequence Diagram

anqichen9856 committed:
Merge pull request #318 from anqichen9856/master

Update UG, DG & PPP

qweiping31415 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main

anqichen9856 committed:
Update documentation

qweiping31415 committed:
delete test

qweiping31415 committed:
delete test

anqichen9856 committed:
Merge pull request #321 from qweiping31415/updateDocs

Update docs

anqichen9856 committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main

* 'master' of https://github.com/AY1920S1-CS2103T-T11-1/main:
  delete test
  delete test
  Updated Sequence Diagram
  Update Documentation

anqichen9856 committed:
Merge pull request #322 from anqichen9856/master

Update docs

briyanii committed:
Added acknowledgements to external libraries
